### ISSUE TITLE
Improve accessibility for screen reader users

### ISSUE THREAD

**Issue by Neurrone** [apache/couchdb-fauxton#1197]
## Expected Behavior

Screen reader users should have full access to the interface.

## Current Behavior

Too many to list, but generally, there are missing labels, and use of `<div>` or `<span>` instead of a more semantically appropriate element. 

## Possible Solution

See above

## Steps to Reproduce (for bugs)

Encountered while following the tutorials:

1. When minimized, navigation links have no labels, so are just read as "link" for each item. The control that toggles minimize/maximize has no role and label.
2. Incoming notifications are not read automatically.

## Your Environment

Latest Fauxton on Apache CouchDB v. 2.3.1
Browser: Firefox 67
Other software used: NVDA screen reader v2019.2
